docs/src/tutorial.md

```diff
@@ -62,6 +62,97 @@ Here is an example using the constrained problem solve:
 stats.solver_specific[:internal_msg]
 ```
 
+## Monitoring optimization with callbacks
+
+You can monitor the optimization process using a callback function. The callback allows you to access the current iterate and constraint violations at each iteration, which is useful for custom stopping criteria, logging, or real-time analysis.
+
+### Callback signature
+
+The callback function must have the following signature:
+
+```julia
+function my_callback(alg_mod, iter_count, problem_ptr, args...)
+    # Your custom code here
+    return true  # return false to stop optimization
+end
+```
+
+### Accessing current iterate information
+
+The `problem_ptr` argument provides access to the current state of the optimization:
+
+- `Ipopt.GetIpoptCurrentIterate(problem_ptr)` returns:
+  - `x`: current primal variables
+  - `z_L`: current multipliers for lower bounds  
+  - `z_U`: current multipliers for upper bounds
+  - `g`: current constraint values
+  - `lambda`: current multipliers for constraints
+
+- `Ipopt.GetIpoptCurrentViolations(problem_ptr)` returns:
+  - `constr_viol`: constraint violation
+  - `dual_inf`: dual infeasibility  
+  - `compl`: complementarity
+
+### Example usage
+
+Here's a complete example showing how to use callbacks to monitor the optimization:
+
+```@example ex4
```

Comment:
The documentation build fails https://github.com/JuliaSmoothOptimizers/NLPModelsIpopt.jl/actions/runs/16833945013/job/47745496477?pr=135#step:5:11